### ISSUE TITLE
FreeBSD 12 build is broken

### ISSUE THREAD

**Issue by tymat** [codahale/bcrypt-ruby#196]
```
gem install bcrypt
Building native extensions. This could take a while...
ERROR:  Error installing bcrypt:
	ERROR: Failed to build gem native extension.

    current directory: /usr/home/appuser/.rvm/gems/ruby-2.6.3/gems/bcrypt-3.1.12/ext/mri
/usr/home/appuser/.rvm/rubies/ruby-2.6.3/bin/ruby -I /usr/home/appuser/.rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0 -r ./siteconf20190430-16026-1k8jamo.rb extconf.rb
creating Makefile

current directory: /usr/home/appuser/.rvm/gems/ruby-2.6.3/gems/bcrypt-3.1.12/ext/mri
make "DESTDIR=" clean

current directory: /usr/home/appuser/.rvm/gems/ruby-2.6.3/gems/bcrypt-3.1.12/ext/mri
make "DESTDIR="
compiling bcrypt_ext.c
In file included from bcrypt_ext.c:2:
./ow-crypt.h:19:14: error: conflicting types for 'crypt_r'
extern char *crypt_r(__CONST char *key, __CONST char *setting, void *data);
             ^
/usr/include/unistd.h:499:7: note: previous declaration is here
char    *crypt_r(const char *, const char *, struct crypt_data *);
         ^
bcrypt_ext.c:17:25: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
            NIL_P(input) ? 0 : RSTRING_LEN(input));
                               ^~~~~~~~~~~~~~~~~~
/usr/home/appuser/.rvm/rubies/ruby-2.6.3/include/ruby-2.6.0/ruby/ruby.h:1000:6: note: expanded from macro 'RSTRING_LEN'
     RSTRING_EMBED_LEN(str) : \
     ^~~~~~~~~~~~~~~~~~~~~~
/usr/home/appuser/.rvm/rubies/ruby-2.6.3/include/ruby-2.6.0/ruby/ruby.h:996:6: note: expanded from macro 'RSTRING_EMBED_LEN'
     (long)((RBASIC(str)->flags >> RSTRING_EMBED_LEN_SHIFT) & \
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bcrypt_ext.c:17:25: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
            NIL_P(input) ? 0 : RSTRING_LEN(input));
                               ^~~~~~~~~~~~~~~~~~
/usr/home/appuser/.rvm/rubies/ruby-2.6.3/include/ruby-2.6.0/ruby/ruby.h:1001:28: note: expanded from macro 'RSTRING_LEN'
     RSTRING(str)->as.heap.len)
     ~~~~~~~~~~~~~~~~~~~~~~^~~
2 warnings and 1 error generated.
*** Error code 1

Stop.
make: stopped in /usr/home/appuser/.rvm/gems/ruby-2.6.3/gems/bcrypt-3.1.12/ext/mri

make failed, exit code 1

Gem files will remain installed in /usr/home/appuser/.rvm/gems/ruby-2.6.3/gems/bcrypt-3.1.12 for inspection.
Results logged to /usr/home/appuser/.rvm/gems/ruby-2.6.3/extensions/x86_64-freebsd-12/2.6.0/bcrypt-3.1.12/gem_make.out
```

**Comment by courtland** [comment]
It looks like this was finally fixed in commit 7644e3684d96fb49f5aa20cb5f547cb1b56b3f38, but an updated gem has not been released yet.

```
test# gem install specific_install
Fetching specific_install-0.3.5.gem
Successfully installed specific_install-0.3.5
Parsing documentation for specific_install-0.3.5
Installing ri documentation for specific_install-0.3.5
Done installing documentation for specific_install after 0 seconds
1 gem installed

test# gem specific_install https://github.com/codahale/bcrypt-ruby.git
/usr/local/bin/git
git installing from https://github.com/codahale/bcrypt-ruby.git
Cloning into '/tmp/d20190501-20959-voaob9'...
remote: Enumerating objects: 1065, done.
remote: Total 1065 (delta 0), reused 0 (delta 0), pack-reused 1065
Receiving objects: 100% (1065/1065), 260.45 KiB | 1.21 MiB/s, done.
Resolving deltas: 100% (503/503), done.
WARNING:  open-ended dependency on rspec (>= 3, development) is not recommended
  if rspec is semantically versioned, use:
    add_development_dependency 'rspec', '~> 3'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: bcrypt
  Version: 3.1.12
  File: bcrypt-3.1.12.gem
Building native extensions. This could take a while...
/usr/local/lib/ruby/site_ruby/2.5/rubygems/ext/builder.rb:183: warning: conflicting chdir during another chdir block
/usr/local/lib/ruby/site_ruby/2.5/rubygems/ext/builder.rb:191: warning: conflicting chdir during another chdir block
Successfully installed
`

**Comment by metalefty** [comment]
Dup of #175, #188. Fixed at #189.
